### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by emqi**
Basic frontend in React with React Router

### CODE REVIEW

**Comment by osiewicz** [comment]
Please squash the commits before submitting them for review.
Here is how:
https://gist.github.com/longtimeago/f7055aa4c3bba8a62197

**Comment by emqi** [comment]
Alright, done!

File: half-lajf-app/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
```
Comment (osiewicz) [inline half-lajf-app/README.md:1]: This is not in the scope of this task. Remove this file.

File: half-lajf-app/public/index.html [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
     <!--
-      manifest.json provides metadata used when your web app is added to the
-      homescreen on Android. See https://developers.google.com/web/fundamentals/engage-and-retain/web-app-manifest/
+      manifest.json provides metadata used when your web app is installed on a
+      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
     -->
-    <link rel="manifest" href="%PUBLIC_URL%/manifest.json">
-    <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
```
Comment (osiewicz) [inline half-lajf-app/public/index.html:12]: Do we need a favicon? (line 12). This causes unnecessary diff.

At the very least, make it so that line 12 shows up correctly as an added line.

File: half-lajf-app/src/App.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React, { Component } from 'react';
+import { BrowserRouter, Switch, Route } from 'react-router-dom'
```
Comment (osiewicz) [inline half-lajf-app/src/App.js:2]: Add new line between global imports and project-specific imports (Python-esque).

File: half-lajf-app/src/components/layout/Navbar.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
```
Comment (osiewicz) [inline half-lajf-app/src/components/layout/Navbar.js:2]: Add a separator line between global imports and user-specific imports.

File: half-lajf-app/src/components/layout/Navbar.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import SignedInLinks from './SignedInLinks'
+import SignedOutLinks from './SignedOutLinks'
+import Sidebar from './Sidebar'
+
+const Navbar = () => {
+  return (
+    <div class="navbar-fixed">
+      <nav>
+        <div class="nav-wrapper grey darken-3">
+        <Sidebar />
```
Comment (osiewicz) [inline half-lajf-app/src/components/layout/Navbar.js]: Add proper indentation.

File: half-lajf-app/src/components/layout/SignedInLinks.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import { NavLink } from 'react-router-dom'
+
+const SignedInLinks = () => {
+  return (
+    <div>
+      <ul id="nav-mobile" class="right">
+        <li><NavLink to='/'>Your profile</NavLink></li>
+        <li><NavLink to='/'><span class="red-text">Sign out</span></NavLink></li>
```
Comment (osiewicz) [inline half-lajf-app/src/components/layout/SignedInLinks.js]: One tag per line please.

File: half-lajf-app/src/components/memes/MemeDetails.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react'
+
+export class MemeDetails extends Component {
+  render() {
+    return (
+      <div className="container section meme-details">
```
Comment (osiewicz) [inline half-lajf-app/src/components/memes/MemeDetails.js:7]: Jesus. Format this stuff please.

File: half-lajf-app/src/components/memes/MemeDetails.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        <div className="row">
+          <div className="col s12 m6 offset-m3">
+          <div className="card z-depth-1">
+            <div className="card-content grey-text text-darken-3">
+              <span style={{fontWeight: "bold"}} className="card-title">Meme title</span>
+              <p>Dodane przez TheWojak2</p>
```
Comment (osiewicz) [inline half-lajf-app/src/components/memes/MemeDetails.js]: Hard-coded value (username).

File: half-lajf-app/src/components/memes/MemeDetails.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+            </div>
+            <div class="card-image">
+              <img src={require('./testMeme.png')} />
+            </div>
+            <div className="container section comments">
+              <p>TUTAJ WLECĄ KOMENTARZE, JESZCZE NIE WIEM JAK ALE WLECĄ
```
Comment (osiewicz) [inline half-lajf-app/src/components/memes/MemeDetails.js]: If you don't know where they're gonna come from then use some kind of placeholder.

File: half-lajf-app/src/components/memes/MemesList.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import React, { Component } from 'react'
+import MemeSummary from './MemeSummary'
+
+const MemesList = () => {
+  return (
+    <div className="meme-list section">
+      <MemeSummary />
```
Comment (osiewicz) [inline half-lajf-app/src/components/memes/MemesList.js:7]: That smells.

File: half-lajf-app/src/index.js [hunk 11]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
```
Comment (osiewicz) [inline half-lajf-app/src/index.js:3]: Add newline between project-specific imports and global imports.

**Review by osiewicz — APPROVED** [review]

File: half-lajf-app/src/components/layout/SignedInLinks.js (hunk 6 above)
Comment (emqi) [reply half-lajf-app/src/components/layout/SignedInLinks.js]: how about you frick off?

File: half-lajf-app/src/components/layout/Navbar.js (hunk 4 above)
Comment (emqi) [reply half-lajf-app/src/components/layout/Navbar.js:2]: no.

File: half-lajf-app/public/index.html (hunk 2 above)
Comment (emqi) [reply half-lajf-app/public/index.html:12]: you really gotta be frickin' bored to dig up this stuff

File: half-lajf-app/src/components/memes/MemeDetails.js (hunk 7 above)
Comment (emqi) [reply half-lajf-app/src/components/memes/MemeDetails.js]: ???

File: half-lajf-app/src/components/layout/Navbar.js (hunk 5 above)
Comment (emqi) [reply half-lajf-app/src/components/layout/Navbar.js]: ???

File: half-lajf-app/src/components/memes/MemeDetails.js (hunk 8 above)
Comment (emqi) [reply half-lajf-app/src/components/memes/MemeDetails.js]: thanks genius

File: half-lajf-app/src/components/memes/MemesList.js (hunk 10 above)
Comment (emqi) [reply half-lajf-app/src/components/memes/MemesList.js]: Liking your own comments is like blowing yourself. Sure, you can do this, but anyone else can do it much better

File: half-lajf-app/src/index.js (hunk 11 above)
Comment (emqi) [reply half-lajf-app/src/index.js:3]: Please. I beg you, stop acting like this is some kind of serious project or I'll eventually lose it all

File: half-lajf-app/src/components/memes/MemesList.js (hunk 10 above)
Comment (osiewicz) [reply half-lajf-app/src/components/memes/MemesList.js]: I'd just like to remind you that you were the one to click the "like" button, lol.

File: half-lajf-app/public/index.html (hunk 2 above)
Comment (osiewicz) [reply half-lajf-app/public/index.html:12]: Welcome to the real world.

File: half-lajf-app/src/components/memes/MemesList.js (hunk 10 above)
Comment (emqi) [reply half-lajf-app/src/components/memes/MemesList.js]: Oh really?
image

Comment (osiewicz) [reply half-lajf-app/src/components/memes/MemesList.js]: Come on, you know what I'm talking about - it happened during one of our classes.

Comment (emqi) [reply half-lajf-app/src/components/memes/MemesList.js]: I might or might not know what you're talking about... The thing is, do you have a proof for everybody else?
